Update troubleshooting guide (#2309)
travis-ci/docs-travis-ci-com: user/enterprise/troubleshooting-guide.md

```

It can happen that organizations are also stuck in the syncing state. Since an organization itself does not have a `is_syncing` flag, all users that do belong to it have to be successfully synced.

> ## GitHub validation errors and ` response_status=422 maximum_number_of_statuses` messages observed in logs

Reviewer: :+1:

> ## GitHub validation errors and ` response_status=422 maximum_number_of_statuses` messages observed in logs

Reviewer: I agree with the changes suggested below.

### The Problem

On every commit made when a build runs, a commit status is created for a given SHA. Due to GitHub’s limitations at 1,000 statuses per SHA and context within a repository, if more than 1000 status are created this leads to a validation error.

> This issue should no longer be present in GitHub Apps integrations but will be possible in Webhooks integrations.

Reviewer 2: What I'm wondering about is if it makes sense to mention what could cause this error to happen?

### Strategy

> Following the reports observed, the current workaround would be to have the user re-sync the account with GitHub to generate a new token and overcome the GitHub limitation. There are two options to do so in Enterprise:

Reviewer 2: ```suggestion
Following the reports observed, the current workaround is to have the user re-sync the account with GitHub to generate a new token and overcome the GitHub limitation. There are two options to do so in Enterprise:
```

> 1) A user can sync via Web UI in profile page and clicking on  `Sync account`

Reviewer 3: In my opinion this should be split into actual sections to really break down the steps. For example:

```suggestion
#### User initiates a sync via their profile in web UI

A user can sync their account with your GitHub instance via the web UI:

1. User accesses the web UI by going to `https://<your-travis-ci-enterprise-domain>` in their browser.
2. In the upper right corner of the page hover over the user icon and select 'Profile' from the dropdown menu.
3. In the upper right corner of the profile page click on 'Sync account'.
```

> 2) A user can ssh into platform instance and run the command:

Reviewer 3: And for the second section:

```suggestion
####  Administrator initiates sync via CLI

An administrator can also initiate a sync on behalf of a user via the `travis` CLI tool on the platform machine:

1. Open a ssh connection to the platform machine.
2. Initiate a sync by running `travis sync_users —logins=<GITHUB-LOGIN>`
```

`travis sync_users  —logins=<GITHUB-LOGIN>`

> **Please note** if `—logins=<GITHUB-LOGIN>    ` is not provided this could trigger a sync on every user. 

Reviewer 2: ```suggestion
**Please note** if `—logins=<GITHUB-LOGIN>    ` is not provided this triggers a sync on every user. Depending on the number of total users on your instance this can take a while and impact production operations
```

```

It can happen that organizations are also stuck in the syncing state. Since an organization itself does not have a `is_syncing` flag, all users that do belong to it have to be successfully synced.

## Logs contain many GitHub API 422 errors

### The Problem

On every commit made when a build runs, a commit status is created for a given SHA. Due to GitHub’s limitations at 1,000 statuses per SHA and context within a repository, if more than 1,000 statuses are created this leads to a validation error.
This issue should no longer be present in GitHub Apps integrations but will be present in Webhooks integrations.

### Strategy

The workaround for this issue is to manually re-sync the user account with GitHub. This will generate a fresh token for the user account that has not reached any GitHub API limits.

There are two options listed below to initiate a sync between your Travis CI Enterprise instance and GitHub instance.

> #### User initiates a sync via their profile in web UI

Reviewer: ```suggestion
#### Sync account from Travis CI web interface
```

> A user can sync their account with your GitHub instance via the web UI:

Reviewer: ```suggestion
To sync your account with your GitHub instance:
```

> 1. User accesses the web UI by going to `https://<your-travis-ci-enterprise-domain>` in their browser.

Reviewer: ```suggestion
1. Open `https://<your-travis-ci-enterprise-domain>`.
```

2. In the upper right corner of the page hover over the user icon and select 'Profile' from the dropdown menu.
3. In the upper right corner of the profile page click on 'Sync account'.

> ####  Administrator initiates sync via CLI

Reviewer: ```suggestion
####  Sync from the CLI with administrator privileges```

> An administrator can also initiate a sync on behalf of a user via the `travis` CLI tool on the platform machine:

Reviewer: ```suggestion
An administrator can also initiate a sync on behalf of someone else via the `travis` CLI tool on the platform machine:
```

1. Open a SSH connection to the platform machine.
2. Initiate a sync by running `travis sync_users —logins=<GITHUB-LOGIN>`

> > If `—logins=<GITHUB-LOGIN>` is not provided then this command will trigger a sync on every user. This could result in long runtimes and may impact production operations if you have a large number of total users on your Travis CI Enterprise instance.

Reviewer: Probably best to move this to *before* the steps to prevent accidents
